clientv3/integration/watch_test.go

```diff
@@ -343,16 +343,45 @@ func putAndWatch(t *testing.T, wctx *watchctx, key, val string) {
 	}
 }
 
-// TestWatchResumeComapcted checks that the watcher gracefully closes in case
+func TestWatchResumeCompacted(t *testing.T) {
+	defer testutil.AfterTest(t)
+	testWatchResumeCompacted(t, nil, nil)
+}
+
+func TestWatchResumeCompactedFrequentDisconnect(t *testing.T) {
```

Comment:
It's non-deterministic by nature since it relies on disconnect behavior. If it's going to be deterministic I'd need to add failpoints to the testing infrastructure. If I wrote a separate test it'd be essentially duplicate what's there now.